hnsendpoint.go

```diff
@@ -26,6 +26,28 @@ type HNSEndpoint struct {
 	IsRemoteEndpoint   bool              `json:",omitempty"`
 }
 
+// Type of Resource Support in ModifySystem
+type SystemType string
+
+// SystemType const
+const (
+	ContainerType      SystemType = "Container"
+	VirtualMachineType SystemType = "VirtualMachine"
+	HostType           SystemType = "Host"
+)
+
+// EndpointAttachDetachRequest is the structure used to send request to the container to modify the system
+// Supported resource types are Network and Request Types are Add/Remove
+type EndpointAttachDetachRequest struct {
+	ContainerId    string     `json:"ContainerId,omitempty"`
+	SystemType     SystemType `json:"SystemType"`
+	CompartmentId  uint16     `json:"CompartmentId,omitempty"`
+	VirtualNicName string     `json:"VirtualNicName,omitempty"`

```

Comment:
same as above (Though I don't have a machine with Go installed in front of me atm to test if VirtualNICName throws a different golint error)